File: src/main/java/com/radar/prospection/api/MissionDto.java
package com.radar.prospection.api;

import com.radar.prospection.domain.Mission;
import com.radar.prospection.domain.MissionStatus;
import com.radar.prospection.domain.Source;

import java.time.LocalDateTime;
import java.util.List;

public record MissionDto(
    Long id,
    Source source,
    String title,
    String company,
    String location,
    Boolean remote,
    Integer tjmMin,
    Integer tjmMax,
    String duration,
    List<String> skills,
    String url,
    MissionStatus status,
    Integer fitScore,
    String fitSummary,
    String decisionMakerHint,
    LocalDateTime detectedAt
) {
    public static MissionDto from(Mission m) {
        return new MissionDto(
            m.getId(), m.getSource(), m.getTitle(), m.getCompany(),
            m.getLocation(), m.getRemote(), m.getTjmMin(), m.getTjmMax(),
            m.getDuration(), m.getSkills(), m.getUrl(), m.getStatus(),
            m.getFitScore(), m.getFitSummary(), m.getDecisionMakerHint(),
            m.getDetectedAt()
        );
    }
}

File: backend/src/main/kotlin/com/radar/prospection/api/MissionDto.kt
package com.radar.prospection.api

import com.radar.prospection.domain.Mission
import com.radar.prospection.domain.MissionStatus
import com.radar.prospection.domain.Source
import java.time.LocalDateTime

data class MissionDto(
    val id: Long?,
    val source: Source?,
    val title: String?,
    val company: String?,
    val location: String?,
    val remote: Boolean?,
    val tjmMin: Int?,
    val tjmMax: Int?,
    val duration: String?,
    val skills: List<String>,
    val url: String?,
    val status: MissionStatus?,
    val fitScore: Int?,
    val fitSummary: String?,
    val decisionMakerHint: String?,
    val detectedAt: LocalDateTime?,
    val favorite: Boolean
) {
    companion object {
        fun from(m: Mission) = MissionDto(
            id = m.id,
            source = m.source,
            title = m.title,
            company = m.company,
            location = m.location,
            remote = m.remote,
            tjmMin = m.tjmMin,
            tjmMax = m.tjmMax,
            duration = m.duration,
            skills = m.skills,
            url = m.url,
            status = m.status,
            fitScore = m.fitScore,
            fitSummary = m.fitSummary,
            decisionMakerHint = m.decisionMakerHint,
            detectedAt = m.detectedAt,
            favorite = m.favorite
        )
    }
}
